token-lending/program/src/state/reserve.rs

```diff
@@ -218,20 +218,19 @@ impl ReserveLiquidity {
         Ok(())
     }
 
-    /// Subtract repay amount from total borrows and return rounded repay value
-    pub fn repay(&mut self, repay_amount: Decimal) -> Result<u64, ProgramError> {
-        let rounded_repay_amount = repay_amount.try_round_u64()?;
-        if rounded_repay_amount == 0 {
-            return Err(LendingError::ObligationTooSmall.into());
-        }
-
+    /// Subtract repay amount from total borrows and add to available liquidity
+    pub fn repay(
+        &mut self,
+        integer_amount: u64,
+        decimal_amount: Decimal,
+    ) -> Result<(), ProgramError> {
         self.available_amount = self
             .available_amount
-            .checked_add(rounded_repay_amount)
+            .checked_add(integer_amount)
             .ok_or(LendingError::MathOverflow)?;
-        self.borrowed_amount_wads = self.borrowed_amount_wads.try_sub(repay_amount)?;
+        self.borrowed_amount_wads = self.borrowed_amount_wads.try_sub(decimal_amount)?;
```

Comment:
If we're adding a whole amount of tokens back into the available liquidity, shouldn't we subtract the same amount of whole tokens from the borrowed amount?  Your tests show that this is working correctly and getting down to 0 on the full repay, which probably answers my question :-)

Reply:
Yeah, good question. The intent of my change was to avoid dust balances by having special logic for when the repay amount is higher than the borrow amount (which has a decimal balance due to interest) such that we switch to paying off the full borrow amount. When repay amount is lower than the borrow amount, the `decimal_amount` you pointed out above is just a whole number. 

> Your tests show that this is working correctly and getting down to 0 on the full repay, which probably answers my question :-)

Haha, yeah that was the goal of the proptests! Glad they were able to speak for themselves :)